Library/brew.sh

```diff
@@ -160,6 +162,39 @@ EOS
   esac
 fi
 
+if [[ -n "$HOMEBREW_ANALYTICS" ]]
+then
+  # User UUID file. Used for Homebrew user counting. Can be deleted and
+  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -r "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(cat "$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+  else
+    HOMEBREW_ANALYTICS_USER_UUID="$(uuidgen)"
+    echo "$HOMEBREW_ANALYTICS_USER_UUID" > "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
```

Comment:
Two things need to consider here
* We need to check `uuidgen` will output UUID version 4 or above for all OS X versions. Because an adversary can extract time and MAC address from UUID version 1. You could check this by `brew install ossp-uuid; uuid -d $(uuidgen)`
* I don't think we should store and reuse uuid. i.e. we should generate uuid per request. Otherwise, uuid itself is an identifier which can be used to track and pin point user.